File: src/main/java/com/bazbatlabs/bricksmash/models/Image.java
package com.bazbatlabs.bricksmash.models;

public final class Image {

    public final int textureId;

    public final float x;
    public final float y;
    public final float w;
    public final float h;

    public Image(float x, float y, float w, float h, Texture texture) {

        float textureWidth = (float)texture.width;
        float textureHeight = (float)texture.height;

        this.x = x / textureWidth;
        this.y = y / textureHeight;
        this.w = w / textureWidth;
        this.h = h / textureHeight;

        this.textureId = texture.id;
    }
}

File: src/main/scala/models/Image.scala
package com.bazbatlabs.bricksmash.models

class Image(tx: Float, ty: Float, tw: Float, th: Float, texture: Texture) {
  val textureId: Int = texture.id

  val x: Float = tx / texture.width
  val y: Float = ty / texture.height
  val w: Float = tw / texture.width
  val h: Float = th / texture.height
}

object Image {
  def apply(tx: Float, ty: Float, tw: Float, th: Float, texture: Texture) =
    new Image(tx, ty, tw, th, texture)
}
